### PR TITLE
python{2,3}Package.apprise: fix missing dependencies

### PR DESCRIPTION
**Pull request by jfroche**
###### Motivation for this change

Add the missing dependencies

ZHF: #97479
@NixOS/nixos-release-managers

fix builds https://hydra.nixos.org/build/127627883 https://hydra.nixos.org/build/127621060 https://hydra.nixos.org/build/127648198

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jfroche** [comment]
Result of `nixpkgs-review pr 100446` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python27Packages.apprise</li>
    <li>python27Packages.gntp</li>
  </ul>
</details>

**Comment by jfroche** [comment]
> Result of `nixpkgs-review pr 100446` [1](https://github.com/Mic92/nixpkgs-review)
> 
> 2 packages built:
> * python27Packages.apprise
> * python27Packages.gntp

why not building python37Packages.apprise and python38Packages.apprise ? Should I change the title to include python3Packages.apprise ?

**Comment by jfroche** [comment]
Result of `nixpkgs-review pr 100446` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>python27Packages.apprise</li>
    <li>python27Packages.gntp</li>
    <li>python37Packages.apprise</li>
    <li>python37Packages.gntp</li>
    <li>python38Packages.apprise</li>
    <li>python38Packages.gntp</li>
  </ul>
</details>

File: pkgs/development/python-modules/gntp/default.nix [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+buildPythonPackage rec {
+  pname = "gntp";
+  version = "1.0.3";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1q6scs8lp84v0aph6b5c9jhv51rhq2vmzpdd38db92ybkq0g597l";
+  };
+
```
Comment (jonringer) [inline pkgs/development/python-modules/gntp/default.nix:11]: There are no tests included. Please try to checkout from source and check if they have unit tests, and try to run them. Unit tests give a good indication that they package has a high degree of validity and correctness given the python package set.

If tests are not available, then please use `pythonImportsCheck` to import the most important modules. This isn't as good as unit tests, but will usually give a good indication of run-time errors. Please see [pythonImportsCheck documentation](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pythonimportscheck) for more information.

```

----------------------------------------------------------------------
Ran 0 tests in 0.000s
```

Comment (jfroche) [reply pkgs/development/python-modules/gntp/default.nix:11]: Indeed there are no testcase for this package. Thank you for taking the time to explain `pythonImportCheck`. Tell me if that's good enough

File: pkgs/development/python-modules/apprise/default.nix [hunk 2]
```diff
@@ -15,7 +15,7 @@ buildPythonPackage rec {
   nativeBuildInputs = [ Babel ];
 
   propagatedBuildInputs = [
-    requests requests_oauthlib six click markdown pyyaml
+    requests requests_oauthlib six click markdown pyyaml gntp sleekxmpp
   ];
```
Comment (jonringer) [inline pkgs/development/python-modules/apprise/default.nix:19]: the two new dependencies are only needed for testing https://github.com/caronc/apprise/blob/f6ce71c65d0c5abfa470978cd719d29e69b497ea/setup.py#L109

please put them in `checkInputs`

File: pkgs/development/python-modules/gntp/default.nix (hunk 1 above)
Comment (jonringer) [reply pkgs/development/python-modules/gntp/default.nix:11]: there are tests: https://github.com/kfdm/gntp/tree/master/test

However, it looks like like you would need a growler instance running. I would just say something like:
```
  # requires a growler service to be running
  doCheck = false;
```

to denote why tests were excluded

**Comment by jonringer** [comment]
If you're new to python packaging through nix, I got a video <https://www.youtube.com/watch?v=jXd-hkP4xnU> which covers most packaging scenarios.

File: pkgs/development/python-modules/gntp/default.nix (hunk 1 above)
Comment (jfroche) [reply pkgs/development/python-modules/gntp/default.nix:11]: There might be manual tests but there are no testcase (iow not using a test framework).

File: pkgs/development/python-modules/apprise/default.nix (hunk 2 above)
Comment (jfroche) [reply pkgs/development/python-modules/apprise/default.nix:19]: not so sure that they are only used in tests: https://github.com/caronc/apprise/blob/v0.8.7/apprise/plugins/NotifyXMPP/SleekXmppAdapter.py
https://github.com/caronc/apprise/blob/v0.8.7/apprise/plugins/NotifyGrowl.py#L37

File: pkgs/development/python-modules/gntp/default.nix (hunk 1 above)
Comment (jfroche) [reply pkgs/development/python-modules/gntp/default.nix:11]: I have added the doCheck attribute

**Comment by jfroche** [comment]
> If you're new to python packaging through nix, I got a video https://www.youtube.com/watch?v=jXd-hkP4xnU which covers most packaging scenarios.

Thank you, will watch it !

File: pkgs/development/python-modules/apprise/default.nix (hunk 2 above)
Comment (jonringer) [reply pkgs/development/python-modules/apprise/default.nix:19]: That looks optional, and generally they aren't included in the base package in nixpkgs; as there's not a convenient way to replicate the "extras" features of pip.

**Comment by risicle** [comment]
nixpkgs-review builds happily for me, macos 10.14

File: pkgs/development/python-modules/apprise/default.nix (hunk 2 above)
Comment (jfroche) [reply pkgs/development/python-modules/apprise/default.nix]: ok no problem, should be fixed

**Comment by jfroche** [comment]
@jonringer this should be ok, maybe you are waiting for something I missed ?

File: pkgs/development/python-modules/gntp/default.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{ lib, buildPythonPackage, fetchPypi, nose }:
+
+buildPythonPackage rec {
+  pname = "gntp";
+  version = "1.0.3";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1q6scs8lp84v0aph6b5c9jhv51rhq2vmzpdd38db92ybkq0g597l";
+  };
```
Comment (jonringer) [inline pkgs/development/python-modules/gntp/default.nix:10]: this should be it's own commit:
```
python3Packages.gntp: init at 1.0.3
```

**Comment by jfroche** [comment]
Result of `nixpkgs-review pr 100446` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>python27Packages.apprise</li>
    <li>python27Packages.gntp</li>
    <li>python37Packages.apprise</li>
    <li>python37Packages.gntp</li>
    <li>python38Packages.apprise</li>
    <li>python38Packages.gntp</li>
  </ul>
</details>

File: pkgs/development/python-modules/gntp/default.nix (hunk 3 above)
Comment (jfroche) [reply pkgs/development/python-modules/gntp/default.nix:10]: true, done

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 100446 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>6 packages built:</summary>  <ul>    <li>python27Packages.apprise</li>    <li>python27Packages.gntp</li>    <li>python37Packages.apprise</li>    <li>python37Packages.gntp</li>    <li>python38Packages.apprise</li>    <li>python38Packages.gntp</li>  </ul></details>
